### PR TITLE
Replace oidc-proxy with oauth2-proxy

### PR DESCRIPTION
**Pull request by alkar**
Connects to https://github.com/ministryofjustice/cloud-platform/issues/633

### CODE REVIEW

File: terraform/cloud-platform-components/prometheus.tf [hunk 1]
```diff
@@ -90,34 +90,54 @@ resource "random_id" "session_secret" {
...
+    issuer_url    = "${data.terraform_remote_state.cluster.oidc_issuer_url}"
+    client_id     = "${auth0_client.monitoring.client_id}"
+    client_secret = "${auth0_client.monitoring.client_secret}"
+    cookie_secret = "${random_id.session_secret.b64_std}"
   }
 }
 
-resource "helm_release" "prometheus_proxy" {
-  name          = "prometheus"
+resource "helm_release" "prometheus-proxy" {
```
Comment (jasonBirchall) [inline terraform/cloud-platform-components/prometheus.tf]: Terraform best practice is to use `_` in resource names as per:
https://www.terraform-best-practices.com/naming#general-conventions

File: terraform/cloud-platform-components/prometheus.tf [hunk 2]
```diff
@@ -126,34 +146,33 @@ resource "helm_release" "prometheus_proxy" {
   }
 }
 
-data "template_file" "alertmanager_proxy" {
-  template = "${file("${path.module}/templates/oidc-proxy.yaml.tpl")}"
+data "template_file" "alertmanager-proxy" {
```
Comment (jasonBirchall) [inline terraform/cloud-platform-components/prometheus.tf]: Same as above best practice.

File: terraform/cloud-platform-components/prometheus.tf [hunk 3]
```diff
@@ -90,34 +90,54 @@ resource "random_id" "session_secret" {
...
-    application_healthcheck      = "enabled"
-    application_healthcheck_port = "/-/healthy"
-    oidc_client_id               = "${data.terraform_remote_state.cluster.oidc_client_id}"
-    oidc_client_secret           = "${data.terraform_remote_state.cluster.oidc_client_secret}"
-    oidc_session_secret          = "${random_id.session_secret.hex}"
+    upstream      = "http://kube-prometheus:9090"
+    hostname      = "prometheus.apps.${data.terraform_remote_state.cluster.cluster_domain_name}"
+    exclude_paths = "^/-/healthy$"
+    issuer_url    = "${data.terraform_remote_state.cluster.oidc_issuer_url}"
+    client_id     = "${auth0_client.monitoring.client_id}"
```
Comment (jasonBirchall) [inline terraform/cloud-platform-components/prometheus.tf]: Is it worth keeping this populated upstream i.e. 
`"${data.terraform_remote_state.cluster.oidc_client_id}"`

**Comment by alkar** [comment]
@jasonBirchall @sablumiah [this](https://github.com/ministryofjustice/cloud-platform-infrastructure/compare/4efebcd9622072991b316a8439b4fdba46696799..ddeb72da7bb4a596c3f383660c420fddc26651d2)

**Pull request merged**